src/iam/iam-base.ts

```diff
@@ -155,6 +155,7 @@ export class IAMBase {
         await this.setAddress();
         this.setDid();
         await this.initChain();
+        await this.checkBalanceForConnection();
```

Comment:
This way user without token will be unable to login. Balance should be checked before creating document